types/artifacts.d.ts

```diff
@@ -743,6 +741,56 @@ declare global {
         /** Column number in the script (0-based). */
         columnNumber: number;
       }
+
+      /** Describes a generic console message. */
+      interface BaseConsoleMessage {
+        /**
+         * What emitted the console message (console API, violation log, etc.).
+         * See subtypes for possible values.
+         */
+        source: string,
+        /** The log level: info, warning, etc. See subtypes for possible values. */
+        level: string,
+        /**
+         * The text printed to the console, as shown on the browser. Note objects printed to the
+         * console will be stringified if primitive or JSON-serializable. Otherwise it will be
+         * a String representation of the object.
+         */
+        text: string;
+        /** Time of the console log in milliseconds since epoch. */
+        timestamp: number;
```

Comment:
are there any sources in the future we can imagine where a timestamp isn't available? just trying to think through changes we wouldn't want to be breaking

Reply:
Not that I'm aware of. I'm not as familiar with devtools but I'd guess that every event would have a timestamp.

Reply:
I suppose worst case scenario we could always just throw the result of `Date.now` when we learned about the event on there to stay backwards compatible :)